File: Javascript/QueueArray.js
class Queue {
    constructor(){
        this.arr = [];
    }

    enqueue = (data) => {
        this.arr.unshift(data);
    }

    dequeue = () => {
        if (this.arr.length === 0){
            console.log("Queue Empty");
        }
        return this.arr.pop();
    }

    peek = () => {
        if (this.arr.length === 0){
            console.log("Queue Empty");
        }
        return this.arr[this.arr.length-1];
    }

    isEmpty = () => {
        if (this.arr.length === 0){
            return true;
        }
        return false;
    }

    size = () => {
        return this.arr.length;
    }

    printList = () => {
        this.arr.forEach((item) => {
            console.log(item);
        });
    }
}

const queue = new Queue();
console.log(queue.isEmpty());
	
queue.enqueue(1);
queue.enqueue(3);
queue.enqueue(5);
queue.enqueue(7);
queue.enqueue(9);

console.log(queue.dequeue());
console.log(queue.peek());
console.log(queue.isEmpty());

File: Python/QueueArray.py
class Queue:
    def __init__(self):
        self.arr = []
    
    def enqueue(self, data):
        self.arr.append(data)
    
    def dequeue(self):
        if self.arr is None:
            print("Stack Empty")
        return self.arr.pop(0)
    
    def peek(self):
        if self.arr is None:
            print("Stack Empty")
        return self.arr[0]

    def isEmpty(self):
        if self.arr is None:
            return True
        return False

    def size(self):
        return len(self.arr)

    def print_list(self):
        for i in self.arr:
            print(i)


queue = Queue()
print(queue.isEmpty())
	
queue.enqueue(1)
queue.enqueue(3)
queue.enqueue(5)
queue.enqueue(7)
queue.enqueue(9)

print(queue.dequeue())
print(queue.peek())
print(queue.isEmpty())
